spec/lib/bitmap_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'bitmap'
+
+RSpec.describe Bitmap do
+  let(:empty_bitmap) { Bitmap.new }
+  let(:bitmap) { Bitmap.new(3, 2) }
+  let(:large_bitmap) { Bitmap.new(6, 5) }
+
+  describe '#initialize' do
+    it 'sets the bitmap with the width and height and all pixels to white' do
+      expect(bitmap.render).to eq("OOO\nOOO")
+      expect(bitmap.width).to eq(3)
+      expect(bitmap.height).to eq(2)
+    end
+  end
+
+  describe '#paint_pixel' do
+    context 'for a bitmap with white pixels' do
+      it 'colors the selected pixel' do
+        bitmap.paint_pixel(3, 2, 'W')
+        expect(bitmap.render).to eq("OOO\nOOW")
```

Comment:
I have a feeling these tests might be a bit brittle as they're altering the bitmaps directly.
These could probably be shared examples as well
```
subject { described_class.new(*args).bitmap }

before { subject.paint_pixel(args) }

context '...' do
  let(:args) { [3, 2, 'W'] }  

  its(:render) { is_expected.to eq('OOO\nOOW')
```